ksp_plugin/interface.cpp

```diff
@@ -406,12 +406,17 @@ void principia__GetVersion(
 bool principia__HasEncounteredApocalypse(
     Plugin* const plugin,
     char const** const details) {
-  journal::Method<journal::HasEncounteredApocalypse> m({{plugin}, {details}});
+  journal::Method<journal::HasEncounteredApocalypse> m({plugin}, {details});
   // Ownership will be transfered to the marshmallow.
-  std::string* const allocated_details = new std::string;
+  std::string details_string;
   bool const has_encountered_apocalypse =
-      CHECK_NOTNULL(plugin)->HasEncounteredApocalypse(allocated_details);
-  *CHECK_NOTNULL(details) = allocated_details->c_str();
+      CHECK_NOTNULL(plugin)->HasEncounteredApocalypse(&details_string);
+  UniqueBytes allocated_details(details_string.size() + 1);
```

Comment:
This seems very complicated.  I don't think you need allocated_details at all: just copy details_string into *details.